src/runtime/io.cpp

```diff
@@ -89,7 +89,14 @@ static obj_res mk_file_not_found_error(b_obj_arg fname) {
 static lean_external_class * g_io_handle_external_class = nullptr;
 
 static void io_handle_finalizer(void * h) {
-    lean_always_assert(fclose(static_cast<FILE *>(h)) == 0);
+    int errnum = fclose(static_cast<FILE *>(h));
+    if (errnum != 0) {
```

Comment:
```suggestion
    if (!fclose(static_cast<FILE *>(h))) {
```
For consistency with the style used elsewhere in the file.

Reply:
errnum is used in the next line